src/map/utils/trustutils.cpp

```diff
@@ -0,0 +1,468 @@
+            trust->vitRank   = (uint8)Sql_GetIntData(SqlHandle, 20);
+            trust->agiRank   = (uint8)Sql_GetIntData(SqlHandle, 21);
+            trust->intRank   = (uint8)Sql_GetIntData(SqlHandle, 22);
+            trust->mndRank   = (uint8)Sql_GetIntData(SqlHandle, 23);
+            trust->chrRank   = (uint8)Sql_GetIntData(SqlHandle, 24);
+            trust->defRank   = (uint8)Sql_GetIntData(SqlHandle, 25);
+            trust->attRank   = (uint8)Sql_GetIntData(SqlHandle, 26);
+            trust->accRank   = (uint8)Sql_GetIntData(SqlHandle, 27);
+            trust->evaRank   = (uint8)Sql_GetIntData(SqlHandle, 28);
+
+            // resistances
+            trust->slashres  = (uint16)(Sql_GetFloatData(SqlHandle, 29) * 1000);
+            trust->pierceres = (uint16)(Sql_GetFloatData(SqlHandle, 30) * 1000);
+            trust->hthres    = (uint16)(Sql_GetFloatData(SqlHandle, 31) * 1000);
+            trust->impactres = (uint16)(Sql_GetFloatData(SqlHandle, 32) * 1000);
+
+            trust->firedef    = 0;
+            trust->icedef     = 0;
+            trust->winddef    = 0;
+            trust->earthdef   = 0;
+            trust->thunderdef = 0;
+            trust->waterdef   = 0;
+            trust->lightdef   = 0;
+            trust->darkdef    = 0;
+
+            trust->fireres    = (uint16)((Sql_GetFloatData(SqlHandle, 33) - 1) * -100);
+            trust->iceres     = (uint16)((Sql_GetFloatData(SqlHandle, 34) - 1) * -100);
+            trust->windres    = (uint16)((Sql_GetFloatData(SqlHandle, 35) - 1) * -100);
+            trust->earthres   = (uint16)((Sql_GetFloatData(SqlHandle, 36) - 1) * -100);
+            trust->thunderres = (uint16)((Sql_GetFloatData(SqlHandle, 37) - 1) * -100);
+            trust->waterres   = (uint16)((Sql_GetFloatData(SqlHandle, 38) - 1) * -100);
+            trust->lightres   = (uint16)((Sql_GetFloatData(SqlHandle, 39) - 1) * -100);
+            trust->darkres    = (uint16)((Sql_GetFloatData(SqlHandle, 40) - 1) * -100);
+
+            g_PTrustList.push_back(trust);
+        }
+    }
+}
+
+void FreeTrustList()
+{
+    g_PTrustIDList.clear();
+}
+
+void SpawnTrust(CCharEntity* PMaster, uint32 TrustID)
+{
+    if (PMaster->PParty == nullptr)
+    {
+        PMaster->PParty = new CParty(PMaster);
+    }
+
+    CTrustEntity* PTrust = LoadTrust(PMaster, TrustID);
+    PMaster->PTrusts.insert(PMaster->PTrusts.end(), PTrust);
+    PMaster->StatusEffectContainer->CopyConfrontationEffect(PTrust);
+    PMaster->loc.zone->InsertTRUST(PTrust);
+    PTrust->Spawn();
+
+    PMaster->PParty->ReloadParty();
+}
+
+CTrustEntity* LoadTrust(CCharEntity* PMaster, uint32 TrustID)
+{
+    CTrustEntity* PTrust = new CTrustEntity(PMaster);
+    Trust_t* trustData   = new Trust_t();
+
+    for (uint32 index = 0; index < g_PTrustList.size(); index++)
+    {
+        if (g_PTrustList.at(index)->trustID == TrustID)
+        {
+            trustData = g_PTrustList.at(index);
+            break;
+        }
+    }
```

Comment:
Unfortunately not, there is a gap in 6000-6003, so the ID's don't all line up perfectly with their position in TrustList. This can be simplified into `std::find`, I'd rather a search than an if/else for the regions either side of the gap

Reply:
To make matters EVEN worse, when SE added Monberaux, they didn't put him at the end of the second grouping of trusts, they put him _in the gap_.  Eugh.

Reply:
Ah yeah, the gap. Good call, then!